File: BiteClub/Models/OrderUser.swift
import Foundation

struct OrderUser: Codable, Hashable, Identifiable {
    var id: UUID
    var userId: UUID
    var orderId: UUID
    var createdAt: Date
    var updatedAt: Date
    
    var order: Order?
    var user: User?
    
    enum CodingKeys: String, CodingKey {
        case id = "id"
        case userId = "user_id"
        case orderId = "order_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case order = "order"
        case user = "user"
    }
}

File: BiteRunr/shared/src/commonMain/kotlin/org/biterunr_studios/biterunr/Models/OrderUser.kt
package org.biterunr_studios.biterunr.Models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.datetime.Instant

@Serializable
data class OrderUser(
    @SerialName("id")
    val id: String,

    @SerialName("user_id")
    val userId: String,

    @SerialName("order_id")
    val orderId: String,
    
    @SerialName("status")
    val status: String,

    @SerialName("created_at")
    val createdAt: Instant,

    @SerialName("updated_at")
    val updatedAt: Instant,

    @SerialName("order")
    val order: Order? = null,

    @SerialName("user")
    val user: Profile
)
